pkg/archive/paths.go

```diff
@@ -67,7 +97,23 @@ func WalkZipFiles(ctx context.Context, path string, walkFn FileWalkFn) (err erro
 	return nil
 }
 
-func WalkTarGzFiles(ctx context.Context, path string, walkFn FileWalkFn) (err error) {
+func TarGzipReader(iReader io.Reader) (*tar.Reader, error) {
+	gzipReader, err := gzip.NewReader(iReader)
+	if err != nil {
+		return nil, err
+	}
+	return tar.NewReader(gzipReader), nil
+}
+
+func TarBzip2Reader(iReader io.Reader) (*tar.Reader, error) {
+	return tar.NewReader(bzip2.NewReader(iReader)), nil
+}
+
+func TarUncompressedReader(iReader io.Reader) (*tar.Reader, error) {
+	return tar.NewReader(iReader), nil
+}
+
+func WalkTarFiles(ctx context.Context, path string, getTarReader TarReaderProvider, walkFn FileWalkFn) error {
```

Comment:
This needs documenting.
Especially we should be commenting about the behaviour between the TarReaderProvider and the path parameters.